### PR TITLE
Fix wrapping and warning on back button. Fix Talk link styles

### PR DESCRIPTION
**Pull request by srallen**
Staging branch URL: https://more-button-style-fixes.pfe-preview.zooniverse.org/

Fixes more from #4570:

- Back button smaller on second step (Penguin Watch, Manatee Chat, Galaxy Zoo), intended?
  - Adds `white-space: nowrap` to the styles for the TalkLink component
- dark theme - Done & Talk filled with blue, Back and Done transparent (Penguin Watch), should all these buttons be transparent, then fill on hover?
  - Added Redux connect and ThemeProvider as well as variant styles for the TalkLink component for light and dark themes.
- Back Button: the warning message extends below the box making it harder to read (tested in Firefox 59.0.2).
  - Added a specific vendor prefixed style for Firefox to get the width correct. Firefox will now use `width: -moz-max-content` and IE and Edge continue to fall back to percentages since they do not support this property yet.
- TalkLink: Does it disable them from the keyboard too? If not, consider adding tabindex="-1" when the link is disabled.
  - Added tabindex="-1"
- TalkLink: when Done & Talk is disabled, and rendered as a span, it still has cursor: pointer as a style.
  - I don't have a solution for this. Using `pointer-events: none` means I cannot style the cursor. It's an either/or styling. I decided that disabling pointer events was more important. I'm open to changing this, though.

@beckyrother I need some design input about the updating Talk link button for the dark theme. The theme PR review had comments that the button seemed out of place with the solid blue styling in the dark theme. I took cues from how the other buttons are styled in the dark theme and made it use a border styling that fills on hover or focus instead. Would you like the blue fill color to be different? Here's an example using Galaxy Zoo (select the star or artifact option): https://more-button-style-fixes.pfe-preview.zooniverse.org/projects/zookeeper/galaxy-zoo/classify?env=production 

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by beckyrother** [comment]
I like the blue outline color! Can we use #104A79 for the fill on hover tho?

**Comment by eatyourgreens** [comment]
>I don't have a solution for this. Using pointer-events: none means I cannot style the cursor. It's an either/or styling. I decided that disabling pointer events was more important. I'm open to changing this, though.

I'm not sure I understand this. The hand cursor is showing up because the Done & Talk has the `cursor: pointer` style in the CSS, even when disabled.

File: app/classifier/components/TaskNavButtons/components/TalkLink/TalkLink.jsx [hunk 1]
```diff
@@ -21,39 +23,78 @@ const commonStyles = `
...
+      dark: `thin solid ${TALK_LINK_BLUE}`,
+      light: `thin solid ${TALK_LINK_BLUE_HOVER}`
+    })};
   }
 `;
 
-export const StyledDisabledTalkPlaceholder = styled.span`
+// can't style the cursor and pointer-events: none simulatenously
+export const StyledDisabledTalkPlaceholder = styled.span.attrs({
+  tabIndex: -1
```
Comment (eatyourgreens) [inline app/classifier/components/TaskNavButtons/components/TalkLink/TalkLink.jsx]: tabIndex only applies to elements that can receive focus. Spans can't receive focus so this isn't necessary. Sorry, my original comment was about disabling links. Spans don't receive pointer events by default either, so I'm not sure why the span would use `pointer-events: none`.

File: app/classifier/components/TaskNavButtons/components/TalkLink/TalkLink.jsx [hunk 2]
```diff
@@ -21,39 +23,78 @@ const commonStyles = `
...
+  border: ${theme('mode', {
+    dark: `thin solid ${TALK_LINK_BLUE}`,
+    light: 'thin solid transparent'
+  })};
+  color: ${theme('mode', {
+    dark: zooTheme.colors.darkTheme.font,
+    light: 'white'
+  })};
   opacity: 0.5;
   pointer-events: none;
```
Comment (eatyourgreens) [inline app/classifier/components/TaskNavButtons/components/TalkLink/TalkLink.jsx]: This shouldn't be needed on a span.

File: app/classifier/components/TaskNavButtons/components/TalkLink/TalkLink.jsx [hunk 3]
```diff
@@ -8,10 +8,12 @@ import { darken, lighten } from 'polished';
...
+// These hasn't been moved into the zoo theme because the button might go away
+const TALK_LINK_BLUE = '#43bbfd';
+const TALK_LINK_BLUE_HOVER = '#69c9fd';
+
 const commonStyles = `
-  background: #43bbfd;
-  border: 0;
-  color: #fff;
+  box-sizing: border-box;
   cursor: pointer;
```
Comment (eatyourgreens) [inline app/classifier/components/TaskNavButtons/components/TalkLink/TalkLink.jsx]: This should only be in the styles for the link element, not the span.

File: app/classifier/components/TaskNavButtons/components/TalkLink/TalkLink.jsx (hunk 1 above)
Comment (eatyourgreens) [reply app/classifier/components/TaskNavButtons/components/TalkLink/TalkLink.jsx]: My fault for not reading the code properly in my first review.

**Pull request merged**